tests/integration_tests.rs

```diff
@@ -670,3 +674,38 @@ fn do_not_panic_regression_tests() {
             .success();
     }
 }
+
+#[test]
+fn do_not_detect_different_syntax_for_stdin_and_files() {
+    let file = "regression_tests/issue_985.js";
+
+    let output_for_file = bat()
+        .arg("--color=always")
+        .arg("--map-syntax=*.js:Markdown")
+        .arg(&format!("--file-name={}", file))
+        .arg("--style=plain")
+        .arg(file)
+        .assert()
+        .success()
+        .get_output()
+        .stdout
+        .clone();
+
+    let output_for_stdin = bat()
+        .arg("--color=always")
+        .arg("--map-syntax=*.js:Markdown")
+        .arg("--style=plain")
+        .arg(&format!("--file-name={}", file))
+        .pipe_stdin(Path::new(EXAMPLES_DIR).join(file))
+        .unwrap()
+        .assert()
+        .success()
+        .get_output()
+        .stdout
+        .clone();
+
+    assert_eq!(
+        from_utf8(&output_for_file).unwrap(),
```

Comment:
couldn't this be `output_for_file.from_utf8().unwrap()`, without the `std::str::from_utf8` import?

Reply:
minor: I try to avoid `unwrap` if possible. In test code, it's fine, but I personally prefer `.expect("…")` with a message.

Reply:
Oh, right. Sorry for the noise :-)